lib/reverse_string.rb

```diff
@@ -1,5 +1,27 @@
 def reverse_characters(str)
+    array1 = []
+    array2 = []
+    
+    str.split(//).each do |c|
```

Comment:
Funnily enough, `str.split(//)` returns an array with each character as an element... which is exactly what your loop does.

That is, you could save some lines of code here by simply writing `array1 = str.split(//)` at the beginning. :)